openstf/pipeline/train_create_forecast_backtest.py

```diff
@@ -42,11 +60,129 @@ def train_model_and_forecast_back_test(
     if training_horizons is None:
         training_horizons = DEFAULT_TRAIN_HORIZONS
 
-    # Call common training pipeline
-    model, report, train_data, validation_data, test_data = train_pipeline_common(
-        pj, modelspecs, input_data, training_horizons, test_fraction=0.15, backtest=True
+    apply_folds = True if n_folds > 1 else False
+
+    models_entire = []
+    valid_data_entire = []
+    train_data_entire = []
+    test_data_entire = []
+
+    forecast = None
+
+    if apply_folds:
+        # prepare data in order to apply nfolds
+        input_data.index = pd.to_datetime(input_data.index)
+        input_data["dates"] = input_data.index.date
+        input_data["random_fold"] = None
+
+        # divide each day in a fold
+        input_data = group_kfold(input_data, n_folds)
+
+        # iterate of the folds, train and forecast for each fold
+        for i in range(n_folds):
+            test_idx = input_data[
+                input_data.index.isin(input_data[input_data.random_fold == i].index)
+            ].index.sort_values()
+            train_idx = input_data[
+                input_data.index.isin(input_data[input_data.random_fold != i].index)
+            ].index.sort_values()
+            test_df = input_data.loc[test_idx, :]
+            train_df = input_data.loc[train_idx, :]
+
+            test_df = TrainFeatureApplicator(
+                horizons=training_horizons, feature_names=modelspecs.feature_names
+            ).add_features(
+                test_df.iloc[:, :-2]
+            )  # ignore the added columns (dates, random_fold)
+
+            (
+                forecast_i,
+                model,
+                train_data,
+                validation_data,
+                test_data,
+            ) = train_model_and_forecast_test_core(
+                pj,
+                modelspecs,
+                train_df.iloc[:, :-2],  # ignore the added columns (dates, random_fold)
+                test_df,
+                training_horizons,
+                apply_folds,
+                test_fraction=0.0,
+            )
+
+            models_entire.append(model)
+            valid_data_entire.append(validation_data)
+            train_data_entire.append(train_data)
+            test_data_entire.append(test_data)
+
+            if i > 0:
```

Comment:
you can get rid of this if statement by defining an empty dataframe with to apropriate columns before the for loop